core-ui/src/components/Predefined/Create/ApiRules/ApiRules.create.js

```diff
@@ -0,0 +1,140 @@
+import React, { useState, useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import * as jp from 'jsonpath';
+
+import { ResourceForm } from 'shared/ResourceForm/ResourceForm';
+
+import { createApiRuleTemplate, createRuleTemplate } from './templates';
+import { useServicesQuery, ServiceDropdown } from './ServiceDropdown';
+import { useGatewaysQuery, GatewayDropdown } from './GatewayDropdown';
+import { HostAndSubdomain } from './HostAndSubdomain';
+import { RuleForm, SingleRuleInput } from './Rules';
+import {
+  findGateway,
+  getGatewayHosts,
+  hasWildcard,
+  validateApiRule,
+} from './helpers';
+
+export function ApiRulesCreate({
+  formElementRef,
+  namespace,
+  onChange,
+  setCustomValid,
+  serviceName,
+}) {
+  const { t } = useTranslation();
+  // queries are moved up here so that the network calls are not doubled
+  const servicesQuery = useServicesQuery(namespace);
+  const gatewaysQuery = useGatewaysQuery(namespace);
+
+  const [subdomain, setSubdomain] = useState('');
+  const [apiRule, setApiRule] = useState(createApiRuleTemplate(namespace));
+
+  // validation
+  useEffect(() => {
+    setCustomValid(validateApiRule(apiRule));
+  }, [apiRule, setCustomValid]);
+
+  // preselect service name when services list loads
+  useEffect(() => {
+    if (servicesQuery.data && serviceName) {
+      const service = servicesQuery.data.find(
+        svc => svc.metadata.name === serviceName,
+      );
+      if (service) {
+        jp.value(apiRule, '$.spec.service.name', serviceName);
+        jp.value(apiRule, '$.spec.service.port', service.spec.ports[0]?.port);
+        setApiRule({ ...apiRule });
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
a total nitpick, but this comment looks strange not aligned with the line it disables :laughing: 
and personally, i'd use this instead:
```
}, [gatewaysQuery.loading]); // eslint-disable-line react-hooks/exhaustive-deps
```
but it's not blocking in any way
(same story with all other similar cases)